src/components/NavBar/NavBar.component.jsx

```diff
@@ -1,33 +1,40 @@
 import React, { useState } from 'react';
+import { useGlobal } from '../../providers/Global';
 import { API_BASE_URL } from '../../utils/constants';
+import { useFetch } from '../../utils/hooks/useFetch';
 import { NavBar as Navigation, SearchFieldWrapper, SearchField, CheckBox, CheckBoxLabel, CheckBoxWrapper, LoginButton, RightDiv } from './styled';
 
-const NavBar = ({ onVideosChanged }) => {
-    const [searchValue, setSearchValue] = useState('elon musk on TED Talks');
-    const [darkMode, setDarkMode] = useState(true);
+const NavBar = () => {
+    const { state, dispatch } = useGlobal();
+    const { searchKeyword, isDarkMode } = state;
+    const [darkMode, setDarkMode] = useState(isDarkMode);
```

Comment:
Here you have the global state and then you define a variable that holds the local state of the component but it is redundant to keep the local state since the global state is coupled to the local variable, you can delete the local state and just use the global
```suggestion
    const { searchKeyword, isDarkMode } = state;
```